src/user/user.service.ts

```diff
@@ -8,10 +10,38 @@ import { UpdateUserDto } from './dto/update-user.dto';
 export class UserService {
   constructor(private readonly prismaService: PrismaService) {}
 
-  async findAll() {
-    return await this.prismaService.user.findMany({
-      include: { Profile: true },
+  async validateUserAutherization(id: string, userId: string) {
+    const userProfile = await this.prismaService.userProfile.findFirst({
+      where: { id },
     });
+
+    if (!userProfile) {
+      throw new NotFoundException(`User Profile with ID ${id} not found`);
+    }
+
+    if (userProfile.userId !== userId) {
+      throw new UnauthorizedException('You are not authorized to perform this action');
+    }
+  }
+
+  async findAll() {
+    try {
+      const users = await this.prismaService.user.findMany({
+        include: { Profile: true },
+      });
+
+      if (!users) {
+        throw new NotFoundException('Users not found');
```

Comment:
most probably it's not an exception. empty array is also valid response it there are no users in db which match the search parameters